cpp/ql/src/semmle/code/cpp/models/implementations/Allocation.qll

```diff
@@ -0,0 +1,260 @@
+  int sizeArg;
+
+  MallocAllocationFunction() {
+    exists(string name |
+      hasGlobalOrStdName(name) and
+      // malloc(size)
+      (name = "malloc" and sizeArg = 0)
+      or
+      hasGlobalName(name) and
+      (
+        // ExAllocatePool(type, size)
+        name = "ExAllocatePool" and sizeArg = 1
+        or
+        // ExAllocatePool(type, size, tag)
+        name = "ExAllocatePoolWithTag" and sizeArg = 1
+        or
+        // ExAllocatePoolWithTagPriority(type, size, tag, priority)
+        name = "ExAllocatePoolWithTagPriority" and sizeArg = 1
+        or
+        // ExAllocatePoolWithQuota(type, size)
+        name = "ExAllocatePoolWithQuota" and sizeArg = 1
+        or
+        // ExAllocatePoolWithQuotaTag(type, size, tag)
+        name = "ExAllocatePoolWithQuotaTag" and sizeArg = 1
+        or
+        // IoAllocateMdl(address, size, flag, flag, irp)
+        name = "IoAllocateMdl" and sizeArg = 1
+        or
+        // IoAllocateErrorLogEntry(object, size)
+        name = "IoAllocateErrorLogEntry" and sizeArg = 1
+        or
+        // MmAllocateContiguousMemory(size, maxaddress)
+        name = "MmAllocateContiguousMemory" and sizeArg = 0
+        or
+        // MmAllocateContiguousNodeMemory(size, minaddress, maxaddress, bound, flag, prefer)
+        name = "MmAllocateContiguousNodeMemory" and sizeArg = 0
+        or
+        // MmAllocateContiguousMemorySpecifyCache(size, minaddress, maxaddress, bound, type)
+        name = "MmAllocateContiguousMemorySpecifyCache" and sizeArg = 0
+        or
+        // MmAllocateContiguousMemorySpecifyCacheNode(size, minaddress, maxaddress, bound, type, prefer)
+        name = "MmAllocateContiguousMemorySpecifyCacheNode" and sizeArg = 0
+        or
+        // MmAllocateNonCachedMemory(size)
+        name = "MmAllocateNonCachedMemory" and sizeArg = 0
+        or
+        // MmAllocateMappingAddress(size, tag)
+        name = "MmAllocateMappingAddress" and sizeArg = 0
+        or
+        // MmAllocatePagesForMdl(minaddress, maxaddress, skip, size)
+        name = "MmAllocatePagesForMdl" and sizeArg = 3
+        or
+        // MmAllocatePagesForMdlEx(minaddress, maxaddress, skip, size, type, flags)
+        name = "MmAllocatePagesForMdlEx" and sizeArg = 3
+        or
+        // MmAllocateNodePagesForMdlEx(minaddress, maxaddress, skip, size, type, prefer, flags)
+        name = "MmAllocateNodePagesForMdlEx" and sizeArg = 3
+        or
+        // LocalAlloc(flags, size)
+        name = "LocalAlloc" and sizeArg = 1
+        or
+        // GlobalAlloc(flags, size)
+        name = "GlobalAlloc" and sizeArg = 1
+        or
+        // HeapAlloc(heap, flags, size)
+        name = "HeapAlloc" and sizeArg = 2
+        or
+        // VirtualAlloc(address, size, type, flag)
+        name = "VirtualAlloc" and sizeArg = 1
+        or
+        // CoTaskMemAlloc(size)
+        name = "CoTaskMemAlloc" and sizeArg = 0
+        or
+        // alloca(size)
+        name = "alloca" and sizeArg = 0
```

Comment:
On GCC, `alloca` is a macro that expands to a call to the function named `__builtin_alloca`.

Reply:
Fixed.